HLTrigger/special/src/HLTHcalLaserMisfireFilter.cc

```diff
@@ -0,0 +1,136 @@
+// -*- C++ -*-
+//
+// Package:    HLTHcalLaserMisfireFilter
+// Class:      HLTHcalLaserMisfireFilter
+// 
+/**\class HLTHcalLaserMisfireFilter HLTHcalLaserMisfireFilter.cc
+
+Description: Filter to select misfires of the HCAL laser.
+
+Implementation:
+Three HBHE RBX's ("bad") do not see the laser signal. Laser events are 
+selected by finding events with a large fraction of digis in HBHE above
+a given threshold that do not have signals in the "bad" RBXes.
+
+For HF events are selected if a large fraction of digis in HF are above 
+a given threshold
+*/
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/Common/interface/RefToBase.h"
+#include "DataFormats/FEDRawData/interface/FEDNumbering.h"
+#include "DataFormats/HcalDigi/interface/HcalCalibrationEventTypes.h"
+#include "EventFilter/HcalRawToDigi/interface/HcalDCCHeader.h"
+#include "HLTrigger/special/interface/HLTHcalLaserMisfireFilter.h"
+
+//
+// constructors and destructor
+//
+HLTHcalLaserMisfireFilter::HLTHcalLaserMisfireFilter(const edm::ParameterSet& config)  : HLTFilter(config) {
+  inputHBHE_            = config.getParameter<edm::InputTag>("InputHBHE");
+  inputHF_              = config.getParameter<edm::InputTag>("InputHF");
+  minFracDiffHBHELaser_ = config.getParameter<double>("minFracDiffHBHELaser");
+  minFracHFLaser_       = config.getParameter<double>("minFracHFLaser");
+  minADCHBHE_           = config.getParameter<int>("minADCHBHE");
+  minADCHF_             = config.getParameter<int>("minADCHF"),
+  testMode_             = config.getUntrackedParameter<bool>("testMode",false);
+
+  inputTokenHBHE_ = consumes<HBHEDigiCollection>(inputHBHE_);
+  inputTokenHF_   = consumes<QIE10DigiCollection>(inputHF_);
+
+ }
+
+HLTHcalLaserMisfireFilter::~HLTHcalLaserMisfireFilter() { }
+ 
+void HLTHcalLaserMisfireFilter::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("InputHBHE",edm::InputTag("source"));
+  desc.add<edm::InputTag>("InputHF",edm::InputTag("source"));
+  desc.add<int>("minADCHBHE",10);
+  desc.add<int>("minADCHF",10);
+  desc.add<double>("minFracDiffHBHELaser",0.3);
+  desc.add<double>("minFracHFLaser",0.3);
+  desc.addUntracked<bool>("testMode",false);
+  descriptions.add("hltHcalLaserMisfireFilter",desc);
+}
+
+bool HLTHcalLaserMisfireFilter::hltFilter(edm::Event& iEvent, const edm::EventSetup&, trigger::TriggerFilterObjectWithRefs & filterproduct) const {
+
```

Comment:
It dioes not make sense that this is an HLTFiltyer, rather than a normal EDFilter, as saveTags() does not make sense for DIGI collections.